canvas_modules/common-canvas/__tests__/common-properties/panels/control-test.js

```diff
@@ -69,3 +69,24 @@ describe("control panel visible and enabled conditions work correctly", () => {
 		expect(controlPanel.hasClass("hide")).to.equal(true);
 	});
 });
+
+describe("control panel classNames applied correctly", () => {
+	let wrapper;
+	beforeEach(() => {
+		const renderedObject = propertyUtils.flyoutEditorForm(panelConditionsParamDef);
+		wrapper = renderedObject.wrapper;
+	});
+
+	afterEach(() => {
+		wrapper.unmount();
+	});
+
+	it("control panels should have custom classname defined", () => {
+		// Total of 27 classnames defined in text panels, but 1 is not in the DOM because of summary
+		expect(wrapper.find("[className$='-group-controls-class']")).to.have.length(26);
```

Comment:
What is this testing above the test below?  I'm a little afraid that since this is so generic we won't really know what is specifically testing in the future.  Same goes for the other tests like this.

Reply:
This is to verify that the number of classNames set in the paramDef matches the actual number of classNames found in the DOM. I agree this doesn't validate the actual panels itself, I will add additional tests to cover each type of panel